tests/test_networkmanager.py

```diff
@@ -514,6 +514,25 @@ def test_settings_secrets(self):
         # Secrets request should contain just vpn section with the secrets in
         self.assertEqual(connection_i.GetSecrets('vpn'), {'vpn': {'secrets': secrets}})
 
+    def test_get_conn_by_uuid(self):
+        uuid = '133d8eb9-6de6-444f-8b37-f40bf9e33226'
+        settings = dbus.Dictionary({
+            'connection': dbus.Dictionary({
+                'id': 'test wireless',
+                'uuid': uuid,
+                'type': '802-11-wireless'}, signature='sv'),
+            '802-11-wireless': dbus.Dictionary({
+                'ssid': dbus.ByteArray('The_SSID'.encode('UTF-8'))}, signature='sv')
+        }, signature='sa{sv}')
+        connectionPath = self.settings.AddConnection(settings)
+        self.assertEqual(self.settings.GetConnectionByUuid(uuid), connectionPath)
+
+        fakeuuid = '123123123213213'
+        try:
```

Comment:
This should use `assertRaises` (see https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises), as otherwise the test would also (wrongly) succceed if calling `GetConnectionByUuid()` with an invalid UID succeeds.

Reply:
I'm happy to use assertRaises, but note that this won't succeed in any case where the uuuid is not what I provided to make it raise, I check that the message output is exactly what I want it to be, any other exception arg would fail.

Reply:
I have to use assertRaaisesRegexp which is the only function that works on both 2.x and 3, but it is deprecated in Python 3...